document/core/binary/types.rst

```diff
@@ -124,13 +124,15 @@ Function Types
 Limits
 ~~~~~~
 
-:ref:`Limits <syntax-limits>` are encoded with a preceding flag indicating whether a maximum is present.
+:ref:`Limits <syntax-limits>` are encoded with a preceding flag indicating whether a maximum is present, and the corresponsing index type.
 
 .. math::
    \begin{array}{llclll}
    \production{limits} & \Blimits &::=&
-     \hex{00}~~n{:}\Bu32 &\Rightarrow& \{ \LMIN~n, \LMAX~\epsilon \} \\ &&|&
-     \hex{01}~~n{:}\Bu32~~m{:}\Bu32 &\Rightarrow& \{ \LMIN~n, \LMAX~m \} \\
+     \hex{00}~~n{:}\Bu64 &\Rightarrow& \I32~~\{ \LMIN~n, \LMAX~\epsilon \} \\ &&|&
+     \hex{01}~~n{:}\Bu64~~m{:}\Bu64 &\Rightarrow& \I32~~\{ \LMIN~n, \LMAX~m \} \\ &&|&
+     \hex{04}~~n{:}\Bu64 &\Rightarrow& \I64~~\{ \LMIN~n, \LMAX~\epsilon \} \\ &&|&
+     \hex{05}~~n{:}\Bu64~~m{:}\Bu64 &\Rightarrow& \I64~~\{ \LMIN~n, \LMAX~m \}
```

Comment:
To match the use site in the rules for memtype and tabletype, you wanna return these as pairs (it, lim).

Reply:
Sorry do you mean use braces around the RHS and the comma between the two values?   I don't see other productions here doing that so maybe I'm misunderstanding?

Reply:
At least a comma, since this is a meta-level pair. There are a couple of places where this occurs, see e.g. the text format rule for [`elemlist`](https://webassembly.github.io/spec/core/text/modules.html#text-elemlist).